astropy/io/fits/connect.py

```diff
@@ -162,7 +163,10 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    # Read time coordinates as Time
+    hdr = FITS_time().read_time(table.header, t)
```

Comment:
I must have terribly overlooked a lot of things now that I've shifted everything to `FITS_time`. Yes, I won't really need separate functions for setting attributes and I definitely won't need an object. Thanks for pointing that out.